emission/analysis/userinput/matcher.py

```diff
@@ -87,6 +87,17 @@ def create_composite_objects(user_id):
 
     last_done_ts = None
     for ct in confirmedTrips:
+        # Before confirmed_place was introduced, we created confirmed_trips without a confirmed_place
+        # For those trips, we will generate a confirmed_place just-in-time and add its ID to the trip
+        # Once every trip has a confirmed_place, we can remove this code
+        if "confirmed_place" not in ct["data"]:
+            cleaned_place = esda.get_entry(esda.CLEANED_PLACE_KEY, ct["data"]["end_place"])
+            confirmed_place_entry = create_confirmed_place_entry(ts, cleaned_place)
+            ts.insert(confirmed_place_entry)
+            ct["data"]["confirmed_place"] = confirmed_place_entry["_id"]
```

Comment:
@JGreenlee how is the `confirmed_place_entry['_id')` filled in?
`Entry.createEntry` fills it in, but you are simply using `new Entry()`
I assumed that you would get the id back from the insert call and use it

Can you verify that the `_id` field is actually non-empty?

```suggestion
            confirmed_place_entry = create_confirmed_place_entry(ts, cleaned_place)
            cpeid = ts.insert(confirmed_place_entry)
            ct["data"]["confirmed_place"] = cpeid
            logging.debug("Setting the confirmed_place key to the newly created id %s" % cpeid)
```

Reply:
`_id` was empty before being added to the timeseries. That's how the first commit was.

The second commit calls ts.insert() first, after which `_id` does exist.

Reply:
The suggestion looks cleaner though. Let's use that.